samples/multi-module/CMakeLists.txt

```diff
@@ -43,7 +43,7 @@ if (NOT CMAKE_BUILD_TYPE)
 endif ()
 
 set(WAMR_BUILD_INTERP 1)
-set(WAMR_BUILD_AOT 0)
+set(WAMR_BUILD_AOT 1)
```

Comment:
I changed `tests/wamr-test-suites/test_wamr.sh` and test the spec cases with AOT/JIT multi module, some errors were reported (case import and linking):
The patch:
```diff
diff --git a/tests/wamr-test-suites/test_wamr.sh b/tests/wamr-test-suites/test_wamr.sh
index 1448e3e1..66fca1bf 100755
--- a/tests/wamr-test-suites/test_wamr.sh
+++ b/tests/wamr-test-suites/test_wamr.sh
@@ -423,7 +423,8 @@ function spec_test()

     # multi-module only enable in interp mode
     if [[ 1 == ${ENABLE_MULTI_MODULE} ]]; then
-        if [[ $1 == 'classic-interp' || $1 == 'fast-interp' ]]; then
+        if [[ $1 == 'classic-interp' || $1 == 'fast-interp'
+              || $1 == 'aot' || $1 == 'jit' ]]; then
```
Then run:
```bash
cd tests/wamr-test-suites
./test_wamr.sh -s spec -t aot -M -b -P
```
Errors were reported:
```
============> run linking failed with a non-zero return code 101

==================== LAST LOG of linking ====================

  target triple: x86_64-pc-linux-gnu
  cpu features:
  opt level:     3
  size level:    3
  output format: AoT file

Starting interpreter for module '/tmp/tmpdhrclega.aot'
Running: ../../../product-mini/platforms/linux/build/iwasm --module-path=/tmp --heap-size=0 -v=0 --repl /tmp/tmpdhrclega.aot
THE FINAL EXCEPTION IS Failed:
  expected: 'OSError(5, 'Input/output error')'
  got: 'AOT module load failed: magic header not detected
'
Compiling AOT to '/tmp/tmp2dkesboq.aot'
Running: ../../../wamr-compiler/build/wamrc --target=x86_64 --cpu=skylake --disable-simd --format=llvmir-opt --disable-llvm-lto -o /tmp/tmp2dkesboq.aot /tmp/tmp9havg4mp.wasm
Invalid file type: expected wasm file but got other
Leaving tempfiles: ['/tmp/tmp_rw7xpfy.wast', '/tmp/tmp9havg4mp.wasm']


==================== LAST LOG END ====================
```
============> run imports failed with a non-zero return code 101

==================== LAST LOG of imports ====================

THE FINAL EXCEPTION IS Failed:
  expected: 'OSError(5, 'Input/output error')'
  got: 'AOT module load failed: magic header not detected
'
Compiling AOT to '/tmp/tmpyx4ffsl_.aot'
Running: ../../../wamr-compiler/build/wamrc --target=x86_64 --cpu=skylake --disable-simd --format=llvmir-opt --disable-llvm-lto -o /tmp/tmpyx4ffsl_.aot /tmp/tmpucdtfi8k.wasm
Create AoT compiler with:
  target:        x86_64
  target cpu:    skylake
  target triple: x86_64-pc-linux-gnu
  cpu features:
  opt level:     3
  size level:    3
  output format: optimized LLVM IR
Compile success, file /tmp/tmpyx4ffsl_.aot was generated.
Leaving tempfiles: ['/tmp/tmp_xjw_334.wast', '/tmp/tmpucdtfi8k.wasm']


==================== LAST LOG END ====================
```
Suggest to enable the multi-module AOT/JIT test in CI, if it is a little complex, we can enable it after this PR is merged.

Reply:
OK, thanks

Reply:
Now, it can build success and the report shows: 
`IN ALL 89 cases: 87 PASS, 2 FAIL, 0 SKIP`
same as main branch.

Reply:
I fixed this 2 failed cases and  the  log shows:
```
THE FINAL EXCEPTION IS Failed:
  expected: 'OSError(5, 'Input/output error')'
  got: '

```
@wenyongh  I am not sure got nothing is right?

Reply:
According to the design of the register_module_list, the sub-module and main-module are same type.
In the spec test, the sub-module's type is wasm, but the main-module is AOT (the default callback for muti-module to load sub-module also is wasm ).
Sooo , the test details is a A.aot as main-module try to load a sub-module named B.wasm, and it return nothing.

Reply:
Not sure whether we can compile the sub module into AOT file in the wamr-test-suites script (seems to be runtest.py) before loading it when we are testing in AOT mode?

Reply:
> I fixed this 2 failed cases and the log shows:
> 
> ```
> THE FINAL EXCEPTION IS Failed:
>   expected: 'OSError(5, 'Input/output error')'
>   got: '
> ```
> 
> @wenyongh I am not sure got nothing is right?

Should be invalid, my understanding is the result should be same as the result of interpreter mode.

Reply:
Hi, the reason for 2 failed case:
1.**imports** :The muti-module's callback for loading sub-module it needs change the type to aot, and then , the spec test's scripts also need to add a function for converting the sub-module's file from  wasm to aot.
2.**linking**: For function, WAMR doesn't consider the function is a import or not when calling a export function [now fixed]. For other types,like table, memory, maybe WAMR doesn't support export/import it in AOT mode ? so  it still failed.

Reply:
https://github.com/bytecodealliance/wasm-micro-runtime/pull/2550

Reply:
@HongxiaWangSSSS Liang submitted PR #2550 for a reference, could you please update the test scripts in this PR according to it?

Reply:
I guess we need filter out the cases with table,global and memory related cases in linking.wast, for these things are not supported yet in this PR

Reply:
I though we've already resolved it in "interpreter multi-module" mode unless there are gaps between "aot multi-module" and "interpreter multi-module".

Reply:
Got the point, we can patch the spec test to pass them.

Reply:
#2550 has been updated. Remove all global,table,memory related cases in *linking.wast*

Reply:
Hi, I have some uncertainty about the spec part dev. I would like to know the update for ignoring the failed cases  and missing  functions in the spec test will be in PR 2550 responsible by @lum1n0us  , or let me update in this PR 2482 ?

Reply:
@HongxiaWangSSSS suggest to update in this PR, and check whether `./test_wamr.sh -s spec -t aot -M -b -P` and `./test_wamr.sh -s spec -t jit -M -b -P` run well?

Reply:
My suggestions are:
- patch *wamr-test-suites* with PR#2500
- run spec test in aot mode
- investigate problems which happen during *imports.wast* and *linking.wast*

Reply:
sorry, I missed this comment and just pushed the code about wamr-test-suites in this pr .